server/data/data-api/src/test/java/org/apache/james/dlp/api/DLPRuleStoreContract.java

```diff
@@ -88,5 +79,22 @@ default void clearShouldOnlyRemoveEntriesOfADomain(DLPRulesStore dlpRulesStore)
 
         assertThat(dlpRulesStore.list(otherDomain)).containsOnly(RULE_2);
     }
+    @Test
+    default void clearShouldOnlyRemovePreviouslyExistingEntries(DLPRulesStore dlpRulesStore) {
+        dlpRulesStore.store(Domain.LOCALHOST, ImmutableList.of(RULE, RULE_2));
 
+        dlpRulesStore.clear(Domain.LOCALHOST);
+
+        dlpRulesStore.store(Domain.LOCALHOST, ImmutableList.of(RULE));
+
+        assertThat(dlpRulesStore.list(Domain.LOCALHOST)).containsOnly(RULE);
+    }
+
+    @Test
+    default void storeShouldNotStoreDuplicates(DLPRulesStore dlpRulesStore) {
+        dlpRulesStore.store(Domain.LOCALHOST, ImmutableList.of(RULE, RULE_2));
+        dlpRulesStore.store(Domain.LOCALHOST, ImmutableList.of(RULE));
```

Comment:
this one is strange. I assumed store to replace all the existing rules by the new ones. So I'd expect here list() to return only RULE

Reply:
Your assumption fits with the previous implementation from @chibenwa which only cares about the last history

Reply:
true ! good catch